pkg/network/dns/stats.go

```diff
@@ -168,22 +170,18 @@ func (d *dnsStatKeeper) ProcessPacketInfo(info dnsPacketInfo, ts time.Time) {
 	d.stats[info.key] = allStats
 }
 
-func (d *dnsStatKeeper) GetNumStats() (int32, int32) {
-	numStats := d.lastNumStats.Load()
-	droppedStats := d.lastDroppedStats.Load()
-	return numStats, droppedStats
-}
-
 func (d *dnsStatKeeper) GetAndResetAllStats() StatsByKeyByNameByType {
 	d.mux.Lock()
 	defer d.mux.Unlock()
 	ret := d.stats // No deep copy needed since `d.stats` gets reset
 	d.stats = make(StatsByKeyByNameByType)
-	log.Debugf("[DNS Stats] Number of processed stats: %d, Number of dropped stats: %d", d.numStats, d.droppedStats)
-	d.lastNumStats.Store(int32(d.numStats))
-	d.lastDroppedStats.Store(int32(d.droppedStats))
-	d.numStats = 0
-	d.droppedStats = 0
+	processedStats := statsTelemetry.processedStats.Load()
+	droppedStats := statsTelemetry.droppedStats.Load()
+	log.Debugf("[DNS Stats] Number of processed stats: %d, Number of dropped stats: %d", processedStats, droppedStats)
+	d.lastProcessedStats = processedStats
+	d.lastDroppedStats = droppedStats
+	statsTelemetry.droppedStats.Delete()
+	statsTelemetry.processedStats.Delete()
```

Comment:
I think deleting the metric here won't allow prometheus to report the value, since that is pull-based.

I believe what you want is something closer to:

```suggestion
	processedStats := statsTelemetry.processedStats.Load()
	droppedStats := statsTelemetry.droppedStats.Load()
	log.Debugf("[DNS Stats] Number of processed stats: %d, Number of dropped stats: %d", processedStats - d.lastProcessedStats, droppedStats - d.lastDroppedStats)
	d.lastProcessedStats = processedStats
	d.lastDroppedStats = droppedStats
```

This does mean you need to track `d.numStats` again, only for purposes of keeping track of the number of entries in `d.stats` and comparing against `d.maxStats`.